### PR TITLE
plugins/cmp-tabnine: move to by-name

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

**Review by MattSturgeon — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergify queue

**Comment by GaetanLepage** [comment]
@mergify queue

**Comment by MattSturgeon** [comment]
```
error: Package ‘tabnine-4.154.0’ in /nix/store/XXX-source/pkgs/development/tools/tabnine/default.nix:48 is not available on the requested hostPlatform:
  hostPlatform.config = "aarch64-unknown-linux-gnu"
  package.meta.platforms = [
    "aarch64-darwin"
    "x86_64-darwin"
    "x86_64-linux"
  ]
  package.meta.badPlatforms = [ ]
, refusing to evaluate.
```

File: tests/test-sources/plugins/by-name/cmp-tabnine/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ lib, pkgs, ... }:
+let
+  platform = pkgs.stdenv.hostPlatform;
+
+  # tabnine is not available on aarch64-linux.
+  doRun = !(platform.isLinux && platform.isAarch64);
+in
+lib.optionalAttrs doRun {
```
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/cmp-tabnine/default.nix:8]: This is the same, right?

```suggestion
lib.optionalAttrs (platform.system != "aarch64-linux") {
```

**Review by MattSturgeon — APPROVED** [review]

File: tests/test-sources/plugins/by-name/cmp-tabnine/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply tests/test-sources/plugins/by-name/cmp-tabnine/default.nix:8]: Yes, considered to be less robust (string comparison ban be wrong and fail silently).

**Comment by GaetanLepage** [comment]
@mergify queue

**Pull request merged**